### ISSUE TITLE
P1: Define and implement all-models HPO hyperparameter grid script

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/minivess-mlops#401]
## Problem

There is no script that runs all model families × all hyperparameter combinations.
Current scripts train the \*best known\* config per family, but there is no structured sweep
for discovering what those best configs should be.

## Target State

\`scripts/train_all_hyperparam_combos.sh\` — runs the full grid search:

\`\`\`bash
# For each model family × each loss candidate × each candidate compute profile:
# → Run training_flow() via scripts/run_training_flow.py
# → Results land in MLflow under minivess_hpo_grid experiment

./scripts/train_all_hyperparam_combos.sh              # full grid
./scripts/train_all_hyperparam_combos.sh --families dynunet,comma_mamba  # subset
./scripts/train_all_hyperparam_combos.sh --losses cbdice_cldice,dice_ce  # subset
./scripts/train_all_hyperparam_combos.sh --debug      # 1-epoch smoke test
\`\`\`

## Prerequisites (must be decided before implementing)

- [ ] **Hyperparameter space**: Which hyperparameters vary per model family?
  - Loss functions: `cbdice_cldice`, `dice_ce`, `dice_ce_cldice`, `betti_matching`?
  - Batch sizes: 1, 2, 4 per model?
  - Learning rates: 1e-3, 3e-4, 1e-4?
  - Num folds: 1 (fast) or 3 (paper)?
- [ ] **Compute budget**: How many combinations before it's unaffordable?
- [ ] **Model families to include**: dynunet, sam3_vanilla, sam3_topolora, sam3_hybrid, comma_mamba, ulike_mamba, mamba_vanilla?
- [ ] **Should this wrap `hpo_flow.py`** (Optuna-based HPO) or be a manual grid?

## Implementation Notes

Once the hyperparameter space is defined, the script is straightforward:
\`\`\`bash
for MODEL in "${MODELS[@]}"; do
    for LOSS in "${LOSSES[@]}"; do
        uv run python scripts/run_training_flow.py \
            --model-family "$MODEL" \
            --loss-name "$LOSS" \
            --trigger-source "train_all_hyperparam_combos.sh"
    done
done
\`\`\`

For proper HPO (Bayesian optimization), use \`hpo_flow.py\` with Optuna instead of
a manual grid — see \`src/minivess/orchestration/flows/hpo_flow.py\`.

## Acceptance Criteria

- [ ] Script created at \`scripts/train_all_hyperparam_combos.sh\`
- [ ] Hyperparameter space documented in \`configs/experiments/hpo_grid.yaml\`
- [ ] Script reads the grid from YAML (no hardcoded combinations in bash)
- [ ] Results land in a dedicated MLflow experiment (\`minivess_hpo_grid\`)
- [ ] \`--dry-run\` flag prints the full grid without training
- [ ] README section added explaining when to use grid vs Optuna HPO

## Related

- \`scripts/train_all_best.sh\` — runs the already-known-best config per family
- \`src/minivess/orchestration/flows/hpo_flow.py\` — Optuna-based HPO (smarter search)
- \`configs/hpo/dynunet_example.yaml\` — HPO search space reference

**Comment by petteriTeikari** [comment]
Completed in PR #508. train_all_hyperparam_combos.sh now reads YAML configs from configs/experiments/, generates Cartesian product via itertools.product(), and launches via docker compose (CLAUDE.md Rules #17/#18). Supports --dry-run, --config, --families, --losses. See configs/experiments/dynunet_grid.yaml and smoke_test.yaml.
